tools/spec-gen-sdk/test/automation/workflowPackage.test.ts

```diff
@@ -0,0 +1,266 @@
+      expect(mockContext.stagedArtifactsFolder).toBe('mocked-join');
+      expect(fs.copyFileSync).not.toHaveBeenCalled();
+    });
+
+    it('should create destination directory and copy artifacts', async () => {
+      mockPackage.artifactPaths = ['mock/artifact.jar'];
+      await workflowPkgSaveSDKArtifact(mockContext, mockPackage);
+      expect(fs.mkdirSync).toHaveBeenCalled();
+      expect(mockPackage.serviceName).toBeDefined();
+    });
+  });
+
+  describe('workflowPkgSaveApiViewArtifact', () => {
+    it('should skip if no apiView artifact path', async () => {
+      mockPackage.apiViewArtifactPath = undefined;
+      await workflowPkgSaveApiViewArtifact(mockContext, mockPackage);
+      expect(fs.copyFileSync).not.toHaveBeenCalled();
+      expect(path.join).not.toHaveBeenCalled();
+    });
+
+    it('should copy apiView artifact to destination', async () => {
+      mockPackage.apiViewArtifactPath = 'mock/apiview.json';
+      await workflowPkgSaveApiViewArtifact(mockContext, mockPackage);
+      expect(fs.mkdirSync).toHaveBeenCalled();
+      expect(fs.copyFileSync).toHaveBeenCalled();
+      expect(path.basename).toHaveBeenCalledWith('mock/apiview.json');
+    });
+  });
+
+  describe('workflowPkgCallInstallInstructionScript', () => {
+    it('should skip if script is not configured', async () => {
+      mockContext.swaggerToSdkConfig.artifactOptions.installInstructionScript = undefined;
+      await workflowPkgCallInstallInstructionScript(mockContext, mockPackage);
+      expect(mockContext.logger.info).toHaveBeenCalledWith('Skip installInstructionScript');
+    });
+
+    it('Handle generate installation instructions faild', async () => {
+      vi.mocked(runSdkAutoCustomScript).mockResolvedValue('failed');
+      await workflowPkgCallInstallInstructionScript(mockContext, mockPackage);
+      expect(mockPackage.installationInstructions).toBe('mockInstallationInstructions');
+      expect(mockPackage.liteInstallationInstruction).toBe('mockLiteInstallationInstruction');
+      expect(mockContext.logger.log).toHaveBeenNthCalledWith(1, 'section', 'Call InstallInstructionScript');
+      expect(mockContext.logger.log).toHaveBeenCalledTimes(2);
+    });
+
+    it('should generate installation instructions', async () => {
+      vi.mocked(runSdkAutoCustomScript).mockResolvedValue('inProgress');
+      vi.mocked(getInstallInstructionScriptOutput).mockReturnValue({
+        full: 'full instructions',
+        lite: 'lite instructions',
+      } as InstallInstructionScriptOutput);
+      await workflowPkgCallInstallInstructionScript(mockContext, mockPackage);
+      expect(mockPackage.installationInstructions).toBe('full instructions');
+      expect(mockPackage.liteInstallationInstruction).toBe('lite instructions');
+      expect(mockContext.logger.log).toHaveBeenCalledWith('section', 'Call InstallInstructionScript');
+      expect(mockContext.logger.log).toHaveBeenCalledTimes(2);
```

Comment:
Can you remove this assertion as we don't want to fail the test when update the loggings?